File: Source/GlueGenerator/PropertyTranslators/TextPropertyTranslator.cpp
#include "TextPropertyTranslator.h"

#include "GlueGenerator/CSScriptBuilder.h"

FTextPropertyTranslator::FTextPropertyTranslator(FCSPropertyTranslatorManager& InPropertyHandlers)
: FBlittableTypePropertyTranslator(InPropertyHandlers, FTextProperty::StaticClass(), "Text")
{
	
}

FString FTextPropertyTranslator::GetNullReturnCSharpValue(const FProperty* ReturnProperty) const
{
	return "default(Text)";
}

void FTextPropertyTranslator::ExportCppDefaultParameterAsLocalVariable(FCSScriptBuilder& Builder, const FString& VariableName, const FString& CppDefaultValue, UFunction* Function, FProperty* ParamProperty) const
{
	Builder.AppendLine(FString::Printf(TEXT("Text %s = Text.None;"), *VariableName));
}

FString FTextPropertyTranslator::GetMarshaller(const FProperty* Property) const
{
	return "TextMarshaller";
}

File: Source/UnrealSharpScriptGenerator/PropertyTranslators/TextPropertyTranslator.cs
using System;
using System.Text;
using EpicGames.UHT.Types;

namespace UnrealSharpScriptGenerator.PropertyTranslators;

public class TextPropertyTranslator : BlittableTypePropertyTranslator
{
    public TextPropertyTranslator() : base(typeof(UhtTextProperty), "FText")
    {
    }
    
    public override bool ExportDefaultParameter => false;
    
    public override string GetNullValue(UhtProperty property)
    {
        return "FText.None";
    }

    public override void ExportCppDefaultParameterAsLocalVariable(GeneratorStringBuilder builder, string variableName, string defaultValue,
        UhtFunction function, UhtProperty paramProperty)
    {
        builder.AppendLine($"FText {variableName} = FText.None;");
    }

    public override string GetMarshaller(UhtProperty property)
    {
        return "TextMarshaller";
    }
}